engine/src/configuration/configuration.cpp

```diff
@@ -22,467 +22,411 @@
-    collision_hacks.front_collision_hack_angle = std::cos(
-            M_PI * GetGameConfig().GetDouble("physics.front_collision_avoidance_hack_angle", 40.0)
-                    / 180.0); // uses default value - must be changed in tandem with constructor!
-    collision_hacks.front_collision_hack_distance = GetGameConfig().GetFloat("physics.front_collision_avoidance_hack_distance",
-            collision_hacks.front_collision_hack_distance);
-    collision_hacks.cargo_deals_collide_damage = GetGameConfig().GetBool("physics.cargo_deals_collide_damage",
-            collision_hacks.cargo_deals_collide_damage);
-
-    // computer substruct
-    computer_config.default_lock_cone = GetGameConfig().GetFloat("physics.lock_cone", computer_config.default_lock_cone);
-    computer_config.default_max_range = GetGameConfig().GetFloat("graphics.hud.radarRange", computer_config.default_max_range);
-    computer_config.default_tracking_cone = GetGameConfig().GetFloat("physics.autotracking", computer_config.default_tracking_cone);
-
-    // fuel substruct
-    fuel.afterburner_fuel_usage =
-            GetGameConfig().GetFloat("physics.AfterburnerFuelUsage", fuel.afterburner_fuel_usage);
-    
-    fuel.fuel_efficiency =
-            GetGameConfig().GetDouble("physics.LithiumRelativeEfficiency_Lithium", fuel.fuel_efficiency);
-    fuel.fuel_equals_warp = GetGameConfig().GetBool("physics.fuel_equals_warp", fuel.fuel_equals_warp);
-    fuel.normal_fuel_usage = GetGameConfig().GetFloat("physics.FuelUsage", fuel.normal_fuel_usage);
-    fuel.reactor_uses_fuel = GetGameConfig().GetBool("physics.reactor_uses_fuel", fuel.reactor_uses_fuel);
-    fuel.vsd_mj_yield = GetGameConfig().GetFloat("physics.VSD_MJ_yield", fuel.vsd_mj_yield);
-    fuel.no_fuel_thrust = GetGameConfig().GetFloat("physics.NoFuelThrust", fuel.no_fuel_thrust);
-    fuel.no_fuel_afterburn = GetGameConfig().GetFloat("physics.NoFuelAfterburn", fuel.no_fuel_afterburn);
-    fuel.variable_fuel_consumption = GetGameConfig().GetBool("physics.VariableFuelConsumption", fuel.variable_fuel_consumption);
-    // Also known as Lithium6constant
-    fuel.deuterium_relative_efficiency_lithium = GetGameConfig().GetFloat("physics.DeuteriumRelativeEfficiency_Lithium", fuel.deuterium_relative_efficiency_lithium);
-    //Fuel Mass in metric tons expended per generation of 100MJ
-    fuel.fmec_factor = GetGameConfig().GetFloat("physics.FMEC_factor", fuel.fmec_factor);
-    fuel.reactor_idle_efficiency = GetGameConfig().GetFloat("physics.reactor_idle_efficiency", fuel.reactor_idle_efficiency);
-    fuel.min_reactor_efficiency = GetGameConfig().GetFloat("physics.min_reactor_efficiency", fuel.min_reactor_efficiency);
-    fuel.ecm_energy_cost = GetGameConfig().GetFloat("physics.ecm_energy_cost", fuel.ecm_energy_cost);
+std::shared_ptr<Configuration> configuration() {
+    static std::shared_ptr<Configuration> kConfiguration = nullptr;
```

Comment:
I think I was the one who introduced this form of singleton for holding the Configuration object. Unfortunately, I don't think my code was quite correct. At least, I don't think it's a true Meyers singleton if it uses a pointer instead of a reference for the static variable.

Hopefully this won't be too difficult to fix in your Json Struct generation code, @royfalk . The Vega Strike Logger class should demonstrate how to implement a Meyers singleton properly. (At least I think I have it right finally.)

Reply:
> I think I was the one who introduced this 

Yes. 

> Hopefully this won't be too difficult to fix in your Json Struct generation code, @royfalk . The Vega Strike Logger class should demonstrate how to implement a Meyers singleton properly. (At least I think I have it right finally.)

I'll take a look.